ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js

```diff
@@ -0,0 +1,512 @@
+        const approving = await contractInstanceToken.methods.approve(tokenAddressRinkebyRouter, amountIn)
+          .send({ from: address })
+        const contractInstanceRouter = await new web3.eth.Contract(routerAbi, tokenAddressRinkebyRouter, {
+          from: address,
+          gasLimit: web3.utils.toHex(10000000),
+          gasPrice: web3.utils.toHex(50e9),
+        });
+        const result = await contractInstanceRouter.methods.swapExactTokensForTokens(amountIn, amountOutMinHex, path, address, deadline)
+          .send({ from: address })
+        return res.status(200).json({ result })
+      } catch (e) {
+        console.error(e);
+        return res.status(500).json({ errors: [e] });
+      }
+    }
+  );
+
+  router.get(
+    "/:address/swap/price", [
+    param("address").trim().isString(),
+    query("tokenFrom").trim().isIn(["dai", "mkr", "uni", "weth", "eth"]),
+    query("tokenTo").trim().isIn(["dai", "mkr", "uni", "weth", "eth"]),
+    query("chain").trim().isIn(["ethereum"]),
+    query("network").trim().isIn(["rinkeby", "mainnet"]),
+  ],
+    async (req, res, next) => {
+      try {
+        const errors = validationResult(req);
+        if (!errors.isEmpty()) {
+          return res.status(400).json({ errors: errors.array() });
+        }
+        const { address = "*" } = req.params;
+        const { amount = "", tokenFrom = "", tokenTo = "", network = "" } = { ...req.query };
+        if (tokenFrom === tokenTo) {
+          res.status(500).json({ error: "tokenFrom and tokenTo cannot be same" })
+        }
+        const tokenTypeFrom = tokenFrom.toString().toUpperCase();
+        const tokenTypeTo = tokenTo.toString().toUpperCase();
+        const networkType = network.toString().toUpperCase();
+
+        let chainIds = {}
+        switch (networkType) {
+          case 'MAINNET':
+            chainIds = ChainId.MAINNET;
+            break;
+          case 'RINKEBY':
+            chainIds = ChainId.RINKEBY;
+            break;
+        }
+        let tokenAddressDai, tokenAddressUniswap, tokenAddressMaker, tokenAddressWrappedEther
+        if (networkType === "MAINNET") {
+          tokenAddressDai = '0x6B175474E89094C44Da98b954EedeAC495271d0F'
+          tokenAddressUniswap = '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984';
+          tokenAddressMaker = '0x9f8F72aA9304c8B593d555F12eF6589cC3A579A2';
+          tokenAddressRouter = '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D'
+          tokenAddressWrappedEther = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
+        }
+        else if (networkType === "RINKEBY") {
+          tokenAddressDai = '0xc7ad46e0b8a400bb3c915120d284aafba8fc4735'
+          tokenAddressUniswap = '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984';
+          tokenAddressMaker = '0xF9bA5210F91D0474bd1e1DcDAeC4C58E359AaD85';
+          tokenAddressRouter = '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D'
+          tokenAddressWrappedEther = '0xc778417E063141139Fce010982780140Aa0cD5Ab';
+        }
```

Comment:
These constants can be lifted out of the logic circuit and into file or constant namespace

Reply:
these constants will be invalid in the coming version , will try to set constants in the share folder in future implementations